assets/src/design-system/theme/global/keyboardOnlyOutline.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Inspired by https://github.com/moxystudio/js-keyboard-only-outlines/blob/master/src/index.js
+
+/**
+ * External dependencies
+ */
+import React, { useEffect, useState } from 'react';
+import { createGlobalStyle } from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import {
+  ACCEPTED_KEYS,
+  KEYBOARD_USER_CLASS,
+  KEYBOARD_USER_SELECTOR,
+} from '../constants/keyboardOnlyOutline';
+
+const OutlineStyles = createGlobalStyle`
+    body:not(${KEYBOARD_USER_SELECTOR}) *:focus {
+        outline: none !important;
+    }
+`;
+
+export const KeyboardOnlyOutline = () => {
+  const [usingKeyboard, setUsingKeyboard] = useState(false);
+  const handleKeydown = (e) => {
+    if (!usingKeyboard && ACCEPTED_KEYS.includes(e.code)) {
+      setUsingKeyboard(true);
+    }
+  };
+
+  const handleMousedown = () => {
+    if (usingKeyboard) {
+      setUsingKeyboard(false);
+    }
+  };
+
+  document.addEventListener('keydown', handleKeydown, true);
+  document.addEventListener('mousedown', handleMousedown, true);
```

Comment:
Let's move 54-55 inside that use effect and ensure it only runs on mount. Currently with no param it will run every time and these event listeners will be added every time.

```
useEffect(() => { ... }, []);
```

An empty array as the dep will make it only run once per lifecycle.

Reply:
nice catch! this has existed for so long in the editor and dashboard i didn't even notice. 🙈  

updated to have the callbacks inside the useEffect so the only dep is `usingKeyboard`. It's being called and cleaning up only when you change from keyboard to mouse or mouse to keyboard.